src/Framework/BuildErrorEventArgs.cs

```diff
@@ -264,6 +315,8 @@ internal override void CreateFromStream(BinaryReader reader, int version)
             columnNumber = reader.ReadInt32();
             endLineNumber = reader.ReadInt32();
             endColumnNumber = reader.ReadInt32();
+
+            helpLink = reader.ReadByte() == 0 ? null : reader.ReadString();
```

Comment:
This should be guarded by a `version` check.

The version is derived here

https://github.com/microsoft/msbuild/blob/d98fde4795add51613433cb5def9c7cbd09f4f15/src/Shared/LogMessagePacketBase.cs#L105-L119

And so we'll have 3 possibilities:

1. .NET Core (some high version, all versions guaranteed to match, can assume it's in the packet)
2. .NET 4.7.2 (version `40`, latest code, can assume it's in the packet)
3. .NET 3.5 (version `20`, message was serialized by _MSBuild assemblies from the .NET Framework_, will not have the field in the packet)

So the deserialization code (here) needs to check `version` before reading the byte, as is done with `projectFile`.

Reply:
I've added a `if (version => 40)` check which seems to be the best approach based on the above information.